modules/core/src/matrix_transform.cpp

```diff
@@ -812,6 +812,64 @@ void flip( InputArray _src, OutputArray _dst, int flip_mode )
         flipHoriz( dst.ptr(), dst.step, dst.ptr(), dst.step, dst.size(), esz );
 }
 
+static void
+flipNDImpl(const uchar* src, const int* shape, const size_t* step, uchar* dst, int axis)
+{
+    int i, total = 1;
+    for (i = 0; i < axis; ++i)
+        total *= shape[i];
+
+    int shape_at_axis = shape[axis];
+    size_t step_at_axis = step[axis];
+    int j, k, idx, t, offset_at_idx, offset;
+    for (i = 0; i < total; ++i)
+    {
+        // retrieve offset for flip location
+        t = i;
+        offset = 0;
+        for (j = axis - 1; j >= 0; --j)
+        {
+            idx = t / shape[j];
+            offset_at_idx = int(t - idx * shape[j]);
+            offset += offset_at_idx * step[j];
+            t = idx;
+        }
```

Comment:
Yes, we need this loop for non-continuous mat.

Bug if we are doing in-place operation, do you know an efficient way to swap two pieces of memory? I use `std::memcpy` to copy from src to dst to avoid the in-place swapping problem.

Reply:
std::swap_ranges of bytes is fine, I think, why do we need extra memory?

Reply:
std::swap_ranges is new to me. Thanks for you enlightened answer. But I found that std::swap_ranges with execution policy is since C++17.

Reply:
I refactor the implementation with std::swap_ranges. Could you review it?